packaging/package/jar_jdk_signer.py

```diff
@@ -28,6 +31,21 @@ def validate_signing_input(input_dir, taco_name, alias, keystore):
     return True
 
 
+def get_user_pwd(alias):
+    """
+    Let user input password for keystore and alias, which will be used as jarsigner subprocess input
+    """
+
+    ks_pwd = getpass.getpass(prompt='Enter keystore password: ', stream=None)
+    alias_pwd = getpass.getpass(prompt='Enter password for alias ' + alias + ":(RETURN if same as keystore password)",
+                                stream=None)
+
+    if alias_pwd == "" or alias_pwd == ks_pwd:
```

Comment:
What is this section doing? I don't see why line 44 and 46 are different.

Reply:
this section is asking user's password for alias and Keystore before spawning jarsigner, if the user's alias password is same with keystore pasword, the two elements in returning tuple are same, otherwise, first one is keystore pwd, the second one is alias pwd.